DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/AnonymizerEngine.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Rules;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    public class AnonymizerEngine
+    {
+        private readonly ILogger _logger = AnonymizerLogging.CreateLogger<AnonymizerEngine>();
+        private readonly AnonymizerSettings _anonymizerSettings;
+        private readonly AnonymizerConfigurationManager _configurationManager;
+        private readonly IAnonymizerRuleFactory _ruleFactory;
+
+        public AnonymizerEngine(string configFilePath = "configuration.json", AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+            : this(AnonymizerConfigurationManager.CreateFromConfigurationFile(configFilePath), anonymizerSettings, ruleFactory)
+        {
+        }
+
+        public AnonymizerEngine(AnonymizerConfigurationManager configurationManager, AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+        {
+            EnsureArg.IsNotNull(configurationManager, nameof(configurationManager));
+
+            _anonymizerSettings = anonymizerSettings ?? new AnonymizerSettings();
+            _configurationManager = configurationManager;
+            _ruleFactory = ruleFactory ?? new AnonymizerRuleFactory(_configurationManager.GetConfiguration(), new DicomProcessorFactory(), new AnonymizerSettingsFactory());
+            _logger.LogDebug("AnonymizerEngine initialized successfully");
+        }
+
+        public void AnonymizeDataset(DicomDataset dataset)
+        {
+            EnsureArg.IsNotNull(dataset, nameof(dataset));
+
+            var context = InitContext(dataset);
+            ValidateInput(dataset);
+            dataset.AutoValidate = false;
+
+            var rules = _ruleFactory.CreateAnonymizationDicomRule(_configurationManager.GetConfiguration().RuleContent);
+            foreach (var rule in rules)
+            {
+                rule.Handle(dataset, context);
+                _logger.LogDebug($"Successfully handle rule {rule.Description}.");
+            }
+
+            ValidateOutput(dataset);
+        }
+
+        private void ValidateInput(DicomDataset dataset)
+        {
+            if (_anonymizerSettings.ValidateInput)
+            {
+                dataset.Validate();
+            }
+        }
+
+        private void ValidateOutput(DicomDataset dataset)
+        {
+            if (_anonymizerSettings.ValidateOutput)
+            {
+                dataset.Validate();
+            }
+        }
```

Comment:
how about merging these two functions into a single one like Validate?

Reply:
like  void Validate(DicomDataset dataset, bool validateIt) ? :)
Maybe simply move the logic to AnonymizeDataset since they're quite simple.

Reply:
If merged, condition expression will in the external functions which seems not elegant?